src/behavior.ts

```diff
@@ -69,27 +69,25 @@ export abstract class Behavior<A> extends Reactive<A, BListener>
   }
   abstract update(t: number): A;
   pushB(t: number): void {
-    if (this.state === State.Push) {
-      const newValue = this.update(t);
-      this.pulledAt = t;
-      if (this.last !== newValue) {
-        this.changedAt = t;
-        this.last = newValue;
-        pushToChildren(t, this);
-      }
+    if (this.changedAt === t) {
+      // This prevents a second push with same timestamp, to be further pushed
+      return;
+    }
+    this.pull(t);
+    if (this.changedAt === t && this.state === State.Push) {
+      pushToChildren(t, this);
```

Comment:
The old `pushB` updated both `changedAt` and `pulledAt`. Would it be better to still do that such that it doesn't become the responsibility of the `pull` methods?